### PR TITLE
[Train] Change default `NCCL_SOCKET_IFNAME` to blacklist `veth`

### LINKED ISSUE #30333: [Train] Ray Train Constants on Multinode causes NCCL Error
### What happened + What you expected to happen

1. Created a cluster using yaml file from on-prem cluster using docker, ran pytorch test script from the website.
    Everything works fine without GPU, but with GPU, NCCL Error occurs.
    GPU only works if I set `--num_workers=1`
    Tried `export NCCL_IB_DISABLE=1` and `export NCCL_P2P_DISABLE=1`, both didnt work
    Tried `export NCCL_DEBUG=WARN` but I cannot see any logs printed on my console

2. works fine with multiple GPU

3. error stack
```
| Trial name               |   # failures | error file                                                                                                      |
|--------------------------+--------------+-----------------------------------------------------------------------------------------------------------------|
| TorchTrainer_96bfe_00000 |            1 | /home/ray/ray_results/TorchTrainer_2022-11-15_19-20-15/TorchTrainer_96bfe_00000_0_2022-11-15_19-20-16/error.txt |
+--------------------------+--------------+-----------------------------------------------------------------------------------------------------------------+

2022-11-15 19:26:36,618	ERROR tune.py:773 -- Trials did not complete: [TorchTrainer_96bfe_00000]
2022-11-15 19:26:36,619	INFO tune.py:778 -- Total run time: 381.25 seconds (380.86 seconds for the tuning loop).
Traceback (most recent call last):
  File "torch_fashion_mnist_example.py", line 155, in <module>
    train_fashion_mnist(num_workers=args.num_workers, use_gpu=args.use_gpu)
  File "torch_fashion_mnist_example.py", line 130, in train_fashion_mnist
    result = trainer.fit()
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/train/base_trainer.py", line 360, in fit
    raise result.error
ray.exceptions.RayTaskError(RuntimeError): ray::_Inner.train() (pid=1141, ip=172.21.36.155, repr=TorchTrainer)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/tune/trainable/trainable.py", line 355, in train
    raise skipped from exception_cause(skipped)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/train/_internal/utils.py", line 54, in check_for_failure
    ray.get(object_ref)
ray.exceptions.RayTaskError(RuntimeError): ray::RayTrainWorker._RayTrainWorker__execute() (pid=1174, ip=172.21.36.155, repr=<ray.train._internal.worker_group.RayTrainWorker object at 0x7fb2dd57d050>)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/train/_internal/worker_group.py", line 31, in __execute
    raise skipped from exception_cause(skipped)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/train/_internal/utils.py", line 129, in discard_return_wrapper
    train_func(*args, **kwargs)
  File "torch_fashion_mnist_example.py", line 106, in train_func
    model = train.torch.prepare_model(model)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/train/torch/train_loop_utils.py", line 124, in prepare_model
    parallel_strategy_kwargs=parallel_strategy_kwargs,
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/train/torch/train_loop_utils.py", line 365, in prepare_model
    model = DataParallel(model, **parallel_strategy_kwargs)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/torch/nn/parallel/distributed.py", line 496, in __init__
    dist._verify_model_across_ranks(self.process_group, parameters)
RuntimeError: NCCL error in: /pytorch/torch/lib/c10d/ProcessGroupNCCL.cpp:845, internal error, NCCL version 2.7.8
ncclInternalError: Internal check failed. This is either a bug in NCCL or due to memory corruption

```

Script run normally until the end

### Versions / Dependencies

version and dependencies should be the same as ray latest-gpu 
OS ubuntu

### Reproduction script

made a cluster with ray cluster launcher, here is the yaml

```
cluster_name: default

docker:
    image: "rayproject/ray-ml:latest-gpu" # You can change this to latest-cpu if you don't need GPU support and want a faster startup
    container_name: "ray_container"
    pull_before_run: True
    run_options:   # Extra options to pass into "docker run"
        - --ulimit nofile=65536:65536
        - -p 8265:8265

provider:
    type: local
    head_ip: x.x.x.x
    worker_ips: [y.y.y.y, z.z.z.z, w.w.w.w, v.v.v.v]

auth:
    ssh_user: x
    ssh_private_key: ~/.ssh/y

min_workers: 4
max_workers: 4
upscaling_speed: 1.0

idle_timeout_minutes: 5

file_mounts: {
}

cluster_synced_files: []
file_mounts_sync_continuously: False

rsync_exclude:
    - "**/.git"
    - "**/.git/**"

rsync_filter:
    - ".gitignore"

initialization_commands: []
ray_cluster -f /home/doors/environment.yaml"]
setup_commands: []

head_setup_commands: []

worker_setup_commands: []

head_start_ray_commands:
    - ray stop
    - ulimit -c unlimited && ray start --head --port=6379 --autoscaling-config=~/ray_bootstrap_config.yaml

worker_start_ray_commands:
    - ray stop
    - ray start --address=$RAY_HEAD_IP:6379
```

train script from https://docs.ray.io/en/latest/train/examples/torch_fashion_mnist_example.html

run command: `python torch_fashion_mnist_example.py --address=x.x.x.x:6379 --num-workers=5 --use-gpu`

### Issue Severity

High: It blocks me from completing my task.

### PR DESCRIPTION
**Pull request by amogkam**
Signed-off-by: amogkam <amogkamsetty@yahoo.com>

Closes https://github.com/ray-project/ray/issues/30333.

Previously, we would set a default NCCL interface whitelist in Ray Train to prioritize ethernet. This is to avoid this issue: https://github.com/anyscale/product/issues/8310.

However, this default whitelist is not fully exhaustive, and prevents users from doing distributed GPU training over wireless: https://github.com/ray-project/ray/issues/30333.

Instead, we change to a blacklist so that NCCL does not use `veth` interface which resolves both issues (thanks @cadedaniel for identifying this!)

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by amogkam** [comment]
Passing test: https://buildkite.com/ray-project/release-tests-pr/builds/26067

**Comment by cadedaniel** [comment]
nice!

From the logs, looks like NCCL is still seeing veth in some of the nodes. Any idea why?

I see 12 worker logs with `veth`, but all 16 have `ens3`
```
$ ag --case-sensitive --literal 'Using [0]ens' | wc -l
      16

$ ag --case-sensitive 'veth' | wc -l
      12
```

```
:job_id:03000000
:actor_name:RayTrainWorker
ip-172-31-201-206:7411:7871 [0] NCCL INFO cudaDriverVersion 11060
ip-172-31-201-206:7411:7871 [0] NCCL INFO Bootstrap : Using ens3:172.31.201.206<0>
ip-172-31-201-206:7411:7871 [0] NCCL INFO NET/Plugin : No plugin found (libnccl-net.so), using internal implementation
ip-172-31-201-206:7411:7885 [0] NCCL INFO Failed to open libibverbs.so[.1]
ip-172-31-201-206:7411:7885 [0] NCCL INFO NET/Socket : Using [0]ens3:172.31.201.206<0> [1]veth01fa248:fe80::343a:e3ff:fe09:aeff%veth01fa248<0>
ip-172-31-201-206:7411:7885 [0] NCCL INFO Using network Socket
ip-172-31-201-206:7411:7885 [0] NCCL INFO Trees [0] 3/-1/-1->2->1 [1] 3/-1/-1->2->1
ip-172-31-201-206:7411:7885 [0] NCCL INFO Channel 00 : 2[1d0] -> 3[1e0] via SHM/direct/direct
ip-172-31-201-206:7411:7885 [0] NCCL INFO Channel 01 : 2[1d0] -> 3[1e0] via SHM/direct/direct
ip-172-31-201-206:7411:7885 [0] NCCL INFO Connected all rings
ip-172-31-201-206:7411:7885 [0] NCCL INFO Channel 00 : 2[1d0] -> 1[1c0] via SHM/direct/direct
ip-172-31-201-206:7411:7885 [0] NCCL INFO Channel 01 : 2[1d0] -> 1[1c0] via SHM/direct/direct
ip-172-31-201-206:7411:7885 [0] NCCL INFO Connected all trees
ip-172-31-201-206:7411:7885 [0] NCCL INFO threadThresholds 8/8/64 | 128/8/64 | 512 | 512
ip-172-31-201-206:7411:7885 [0] NCCL INFO 2 coll channels, 2 p2p channels, 2 p2p channels per peer
ip-172-31-201-206:7411:7885 [0] NCCL INFO comm 0x7f4cd4008ec0 rank 2 nranks 16 cudaDev 0 busId 1d0 - Init COMPLETE
ip-172-31-201-206:7411:7887 [0] NCCL INFO [Service thread] Connection closed by localRank 2
ip-172-31-201-206:7411:7411 [0] NCCL INFO comm 0x7f4cd4008ec0 rank 2 nranks 16 cudaDev 0 busId 1d0 - Abort COMPLETE

```

**Review by cadedaniel — APPROVED** [review]
discussed offline. we're not sure if that line is printed before or after blacklist. we should catch it in nightly if the order is nondeterministic.

can we file an issue against product to remove the virtual ethernet interface? then Ray Train can get out of the way of users completely here (default to NCCL behavior).

File: python/ray/train/constants.py [hunk 1]
```diff
@@ -68,11 +68,8 @@
 # as Trainable)
 DISABLE_LAZY_CHECKPOINTING_ENV = "TRAIN_DISABLE_LAZY_CHECKPOINTING"
 
-# Default NCCL_SOCKET_IFNAME.
-# Use ethernet when possible.
-# NCCL_SOCKET_IFNAME does a prefix match so "ens3" or "ens5" will match with
-# "en".
-DEFAULT_NCCL_SOCKET_IFNAME = "en,eth,bond"
+# Blacklist virtualized networking.
+DEFAULT_NCCL_SOCKET_IFNAME = "^lo,docker,vethc"
```
Comment (cadedaniel) [inline python/ray/train/constants.py:72]: oh actually the reason is that it should be `veth` not `vethc`. `c` happened to be the first hex character of the ID I sent you.

that also explains why it was present in only 12 or the 16 -- the other 4 must have had `c` as the first character of the id.

**Review by cadedaniel — CHANGES_REQUESTED** [review]

File: python/ray/train/constants.py (hunk 1 above)
Comment (amogkam) [reply python/ray/train/constants.py:72]: Ah good catch...then now I'm wondering why this passed. Maybe this is not needed at all 🤔. Let me run the test with this default removed.

**Comment by amogkam** [comment]
fails without it: https://buildkite.com/ray-project/release-tests-pr/builds/26088#0185d2ae-4fae-47ee-b8f6-671cdc492ece

**Comment by amogkam** [comment]
test is passing here: https://buildkite.com/ray-project/release-tests-pr/builds/26227#0185e07f-407c-4067-bc3f-40da248cd547

**Review by cadedaniel — APPROVED** [review]
Can we create an issue against product to remove the virtual network interface?

**Review by krfricke — COMMENTED** [review]
Quick question, the title says we blacklist `vethc` - instead we ~whitelist~ `veth`. Is that what we want? Edit: Sorry misread the line

**Comment by amogkam** [comment]
Updated the title-- it should be `veth`

**Review by krfricke — APPROVED** [review]
Thanks!

**Pull request merged**